packages/flutter_tools/lib/src/cache.dart

```diff
@@ -1796,7 +1796,20 @@ class ArtifactUpdater {
       final Directory destination = location.childDirectory(
         tempFile.fileSystem.path.basenameWithoutExtension(tempFile.path)
       );
-      ErrorHandlingFileSystem.deleteIfExists(destination, recursive: true);
+      try {
+        ErrorHandlingFileSystem.deleteIfExists(
+          destination,
+          recursive: true,
+        );
+      } on FileSystemException catch (error) {
+        if (_platform.isWindows && error.osError.errorCode == 32) {
```

Comment:
nit: Having a named constant for the `32` would be good, like you had in another patch set.
```
const int kSharingViolation = 32;
```

Reply:
done